src/tests/unittests/validation/RenderPassValidationTests.cpp

```diff
@@ -37,6 +37,119 @@ TEST_F(RenderPassValidationTest, OneSubpassOneAttachment) {
+        .SetAttachmentCount(1)
+        .AttachmentSetFormat(0, nxt::TextureFormat::R8G8B8A8Unorm)
+        .GetResult();
+
+    AssertWillBeError(device.CreateRenderPassBuilder())
+        .SetSubpassCount(1)
+        .SetAttachmentCount(1)
+        .AttachmentSetFormat(0, nxt::TextureFormat::R8G8B8A8Unorm)
+        // Test AttachmentSetFormat slot out of bounds
+        .AttachmentSetFormat(1, nxt::TextureFormat::R8G8B8A8Unorm)
+        .GetResult();
+
+    AssertWillBeError(device.CreateRenderPassBuilder())
+        .SetSubpassCount(1)
+        .SetAttachmentCount(1)
+        .AttachmentSetFormat(0, nxt::TextureFormat::R8G8B8A8Unorm)
+        // Test AttachmentSetColorLoadOp slot out of bounds
+        .AttachmentSetColorLoadOp(1, nxt::LoadOp::Clear)
+        .GetResult();
+
+    AssertWillBeError(device.CreateRenderPassBuilder())
+        .SetSubpassCount(1)
+        .SetAttachmentCount(1)
+        .AttachmentSetFormat(0, nxt::TextureFormat::R8G8B8A8Unorm)
+        // Test AttachmentSetDepthStencilLoadOps slot out of bounds
+        .AttachmentSetDepthStencilLoadOps(1, nxt::LoadOp::Clear, nxt::LoadOp::Clear)
+        .GetResult();
+
+    AssertWillBeError(device.CreateRenderPassBuilder())
+        .SetSubpassCount(1)
+        .SetAttachmentCount(1)
+        .AttachmentSetFormat(0, nxt::TextureFormat::R8G8B8A8Unorm)
+        // Test SubpassSetColorAttachment attachment slot out of bounds
+        .SubpassSetColorAttachment(0, 0, 1)
+        .GetResult();
+
+    AssertWillBeError(device.CreateRenderPassBuilder())
+        .SetSubpassCount(1)
+        .SetAttachmentCount(1)
+        .AttachmentSetFormat(0, nxt::TextureFormat::R8G8B8A8Unorm)
+        // Test SubpassSetDepthStencilAttachment attachment slot out of bounds
+        .SubpassSetDepthStencilAttachment(0, 1)
+        .GetResult();
+}
+
+// Test for subpass arguments out of bounds
+TEST_F(RenderPassValidationTest, SubpassOutOfBounds) {
+    // Control case
+    AssertWillBeSuccess(device.CreateRenderPassBuilder())
+        .SetSubpassCount(1)
+        .SetAttachmentCount(1)
+        .AttachmentSetFormat(0, nxt::TextureFormat::R8G8B8A8Unorm)
+        .SubpassSetColorAttachment(0, 0, 0)
+        .GetResult();
+
+    AssertWillBeError(device.CreateRenderPassBuilder())
+        .SetSubpassCount(1)
+        .SetAttachmentCount(1)
+        .AttachmentSetFormat(0, nxt::TextureFormat::R8G8B8A8Unorm)
+        .SubpassSetColorAttachment(0, 0, 0)
+        // Test SubpassSetColorAttachment subpass out of bounds
+        .SubpassSetColorAttachment(1, 0, 0)
+        .GetResult();
+
+    AssertWillBeError(device.CreateRenderPassBuilder())
+        .SetSubpassCount(1)
+        .SetAttachmentCount(1)
+        .AttachmentSetFormat(0, nxt::TextureFormat::D32FloatS8Uint)
+        // Test SubpassSetDepthStencilAttachment subpass out of bounds
+        .SubpassSetDepthStencilAttachment(1, 0)
+        .GetResult();
+}
+
+TEST_F(RenderPassValidationTest, SubpassAttachmentWrongFormat) {
```

Comment:
nit: comment as to what this tests before TEST_F. How about calling the test SubpassAttachmentWrongAspect?